src/DTMFSender.ts

```diff
@@ -0,0 +1,124 @@
+class DTMFSender {
+  private _outputNode: MediaStreamAudioDestinationNode;
+  private _outputStream: MediaStream;
+  private _gain: GainNode;
+  private _filter: BiquadFilterNode;
+  private _sourceNode: MediaStreamAudioSourceNode;
+  private _osc1: OscillatorNode;
+  private _osc2: OscillatorNode;
+  private _toneDuration: number = 1500;
+  private _toneGap: number = 70;
+  private _tone: string = "";
+  private _playing: boolean = false;
+
+  private _dtmfFreq: Map<string, Array<number>> = new Map([
+    ["1", [1209, 697]],
+    ["2", [1336, 697]],
+    ["3", [1477, 697]],
+    ["a", [1633, 697]],
+    ["4", [1209, 770]],
+    ["5", [1336, 770]],
+    ["6", [1477, 770]],
+    ["b", [1633, 770]],
+    ["7", [1209, 852]],
+    ["8", [1336, 852]],
+    ["9", [1477, 852]],
+    ["c", [1633, 852]],
+    ["*", [1209, 941]],
+    ["0", [1336, 941]],
+    ["#", [1477, 941]],
+    ["d", [1633, 941]],
+  ]);
+
+  constructor(sender: RTCRtpSender) {
+    var audioCtx = new AudioContext();
+    this._outputNode = audioCtx.createMediaStreamDestination();
+    this._outputStream = this._outputNode.stream;
+
+    if (sender.track) {
+      var inputStream: any = new MediaStream([sender.track]);
+      var rtpSender: any = sender;
+    } else {
+      var inputStream: any = sender;
+      this._outputStream = this._outputStream;
+      var rtpSender: any = null;
+    }
+
+    this._sourceNode = audioCtx.createMediaStreamSource(inputStream);
+    this._sourceNode.connect(this._outputNode);
+
+    this._osc1 = audioCtx.createOscillator();
+    this._osc1.connect(this._outputNode);
+    this._osc1.type = 'sine';
```

Comment:
should 'sine' be a constant or enum?  are there other options?

Reply:
Actually, looks like Typescript takes care of that. This value _is_ from an enum (`OscillatorType`) and I cannot set it to anything but the strings allowed in the enum.